lib/react_client.dart

```diff
@@ -491,12 +491,16 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
         component.adapter = new JsComponent2Adapter(jsThis: jsThis);
         // Return the component so that the JS proxying component can store it,
         // avoiding an interceptor lookup.
+        component.init();
+
+        if (jsThis.dartComponent != null && jsThis.dartComponent.state != null) {
+          jsThis.setState(new JsBackedMap.from(jsThis.dartComponent.state));
```

Comment:
I don't think we want to do `setState` since that will result in additional state transition. I think you want something like the implementation on the ticket:
```dart
// Sync the JS component's state the component's state
if (component.state != null) {
  jsThis.state = new JsBackedMap.from(component.state).jsMap;
}
```

Reply:
We'll also want to add tests that initializing state from `init` works as expected